### PR TITLE
replace controlTypes.STATE_* with controlTypes.State.*

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:

Part of #12549 

### Summary of the issue:

`controlTypes.STATE_*` are deprecated, to be replaced by `controlTypes.State.*`

### Description of how this pull request fixes the issue:

* replaces all instances of `controlTypes.STATE_` with `controlTypes.State.` across .py files
* search and manually replace final STATE_ constant usages 
     * Searched using the regex `[^\.]\bSTATE_`, which finds STATE_ instances
where:
          * it is not of the form `<Foo>.STATE_ `(as oleacc.STATE_* exists)
          * STATE_ is the prefix of the symbol (as IA2_STATE_* exists)

### Testing strategy:

Manual testing
* Confirm the above steps will not introduce issues, and cover all cases.
* Reproduce those steps and perform a diff with this branch

Existing unit tests exist for controlTypes.
New unit tests are written to ensure displayString and negativeDisplayString works

Existing system tests will work as a basic smoke test for this core feature

### Known issues with pull request:

The lint check is likely to fail

### Change log entries:

None needed

### Code Review Checklist:

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [x] System (end to end) tests.
- [x] Manual testing.
- [x] User Documentation.
- [x] Change log entry.
- [x] Context sensitive help for GUI changes.
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers

### CODE REVIEW

File: source/appModules/powerpnt.py [hunk 1]
```diff
@@ -407,18 +407,18 @@ def _get_firstChild(self):
...
+		self._isHandlingSelectionChange=True
+		try:
 			obj=self.selection
 			if not obj:
 				obj=IAccessible(windowHandle=self.windowHandle,IAccessibleObject=self.IAccessibleObject,IAccessibleChildID=self.IAccessibleChildID)
 			if obj and obj!=eventHandler.lastQueuedFocusObject:
-				eventHandler.queueEvent("gainFocus",obj)
-		finally:
+				eventHandler.queueEvent("gainFocus",obj)
+		finally:
```
Comment (feerrenrut) [inline source/appModules/powerpnt.py:421]: Looks like some newline changes

**Review by feerrenrut — COMMENTED** [review]
Overall looks good.

I've been thinking "how do we prevent new code from using the old `STATE_*` constants?" Perhaps we could extend the lint check (which already creates a diff), then regex on that diff looking for `STATE_*` constants.

**Comment by seanbudd** [comment]
>I've been thinking "how do we prevent new code from using the old STATE_* constants?" Perhaps we could extend the lint check (which already creates a diff), then regex on that diff looking for STATE_* constants.

How would we determine `STATE_*` constant usage? Right now it is manual confirmation (with a regex guide).

Once this is merged in, we could replace the mapping `STATE_* = State.*` with the following idea from @leonardder https://github.com/nvaccess/nvda/pull/12649#issuecomment-880444662, and it would be less noisy. This is a suggested usage of [PEP-562](https://www.python.org/dev/peps/pep-0562/).

```python
def __getattr__(name: str) -> Any:
	if name.startswith("STATE_"):
		log.error("Deprecated constant used")
		return getattr(State, name.split("_", 1)[-1]
	return globals()[name]
```

**Comment by lukaszgo1** [comment]
> 
> 
> > I've been thinking "how do we prevent new code from using the old STATE_* constants?" Perhaps we could extend the lint check (which already creates a diff), then regex on that diff looking for STATE_* constants.
> 
> How would we determine `STATE_*` constant usage? Right now it is manual confirmation (with a regex guide).

Cannot we just search for every `STATE_*` in the diff? I agree that is not very efficient but much better than the alternative below.

> 
> Once this is merged in, we could replace the mapping `STATE_* = State.*` with the following idea from @leonardder [#12649 (comment)](https://github.com/nvaccess/nvda/pull/12649#issuecomment-880444662), and it would be less noisy. 

Less noisy for core but horribly annoying for add-ons making Alphas pretty unusable - imagine getting error sound every once in a while just because many add-ons are not going to use new enum's before 2022.1.

**Comment by leonardder** [comment]
> > Once this is merged in, we could replace the mapping `STATE_* = State.*` with the following idea from @leonardder [#12649 (comment)](https://github.com/nvaccess/nvda/pull/12649#issuecomment-880444662), and it would be less noisy.
> 
> Less noisy for core but horribly annoying for add-ons making Alphas pretty unusable - imagine getting error sound every once in a while just because many add-ons are not going to use new enum's before 2022.1.

Not noisy at all if we lower the error level a bit.

**Comment by lukaszgo1** [comment]
> Not noisy at all if we lower the error level a bit.

That is an option but this means that usage of the oldd constants would be harder to check for new PR's. Essentially looking for them in the diff seems to be a better option than inspecting  the log as the latter solution cannot be automated.

**Comment by seanbudd** [comment]
>Cannot we just search for every STATE_* in the diff? I agree that is not very efficient but much better than the alternative below.

There could be false positives here, eg "oleacc.STATE_*" and "IA2_STATE_*".

>Not noisy at all if we lower the error level a bit.

Or if we only perform an error beep on the first runtime usage of a particular constant or group of constants.

**Comment by feerrenrut** [comment]
>How would we determine STATE_* constant usage? Right now it is manual confirmation (with a regex guide).

By running a regex against the PR diff (Only lines changed, the same as the linter does). 

> There could be false positives here, eg "oleacc.STATE_*" and "IA2_STATE_*".

- Tighten up the regex
- Have a way to exclude a line / or mark as a false positive.

**Review by feerrenrut — APPROVED** [review]

**Comment by seanbudd** [comment]
@feerrenrut - this will require a force merge as the lint check will not pass

**Comment by seanbudd** [comment]
> * Have a way to exclude a line / or mark as a false positive.

I think we can do both, and something like marking it with a comment `# ignore deprecation warning`.

Would adding the warnings/errors be a good idea too? I think it would be better to move to the PEP standard, and update  `deprecations.md`.

**Pull request merged**